### ISSUE TITLE
unused_mut warning incorrectly shown for cfg-gated mutable variable

### ISSUE THREAD

**Issue by rinntaya** [rust-lang/rust-analyzer#20292]
#### Describe the bug:
When using **#[cfg(not(debug_assertions))]** to conditionally mutate a variable in release builds only, rust-analyzer shows a false-positive **unused_mut warning** in VSCode. However, this mutation is only used in non-debug builds, and the mutability is necessary in release mode. Although mut code is not accessible in debug mode, removing mut as suggested causes a compile error in release mode.

**For example the following code**
```rs
fn main() {
    let mut x = 1;
    #[cfg(not(debug_assertions))]
    {
        x = 2;
    }
    println!("{}", x);
}
```
<img width="1348" height="1115" alt="Image" src="https://github.com/user-attachments/assets/e37c644a-04b6-49d5-9a34-d37adbd4fc8b" />

**lsp key information**
```json
{
  "message": "variable does not need to be mutable",
  "relatedInformation": [
    {
      "message": "remove this `mut`"
    }
  ]
}
```

#### Environment:
OS: Windows 11 24H2
VSCode version: 1.102.1
rust-analyzer version: 0.3.2547
Rust version: rustc 1.88.0 (6b00bc388 2025-06-23)

**Comment by ChayimFriedman2** [comment]
That's not even a rust-analyzer warning, it's a rustc warning and the same warning will show if you `cargo check`. The usual way to cope with that is to conditionally declare the variable mutable:
```rust
#[cfg(not(debug_assertions))]
let mut x = 1;
#[cfg(debug_assertions)]
let x = 1;
```

**Comment by rinntaya** [comment]
Sorry, I'm not familiar with rust enough to think it's a rust-analyzer bug ᯠ  _   ̫  _ ̥ ᯄ ੭
~~>_<~~ I forgot to check everything before submitting the issue, sorry for the delay
